src/pykeen/nn/node_piece/anchor_search.py

```diff
@@ -222,6 +222,152 @@ def __call__(self, edge_index: numpy.ndarray, anchors: numpy.ndarray, k: int) ->
         return self.select(pool=pool, k=k)
 
 
+class SparseBFSSearcher(ScipySparseAnchorSearcher):
+    """Find closest anchors using :mod:`torch_sparse` on a GPU."""
+
+
+    @staticmethod
+    def create_adjacency(
+        edge_index: numpy.ndarray,
+    ) -> torch.tensor:
+        """
+        Create a sparse adjacency matrix (in the form of the edge list) from a given edge index.
+
+        :param edge_index: shape: (2, m)
+            the edge index
+
+        :return: shape: (2, 2m + n)
+            edge list with inverse edges and self-loops
+        """
+
+        # infer shape
+        num_entities = edge_index.max().item() + 1
+        edge_index = torch.tensor(edge_index, dtype=torch.long)
```

Comment:
`torch.as_tensor` has some nice properties, e.g., avoiding to copy when the storage can be shared

https://pytorch.org/docs/stable/generated/torch.as_tensor.html#torch.as_tensor

Reply:
https://github.com/pykeen/pykeen/pull/990/commits/36a257a511a5e89b361250cd2fddac2d5960507a